src/main.cpp

```diff
@@ -0,0 +1,1616 @@
+            semaphore_image_available,
+            VK_NULL_HANDLE,
+            &image_index);
+    if (VK_SUCCESS != result) {
+        throw std::runtime_error(
+            "Failed to retrieve index of next available presentable image");
+    }
+
+    vkResetCommandBuffer(command_buffer, flags);
+
+    record_command_buffer(
+            command_buffer,
+            image_index,
+            render_pass,
+            swapchain_framebuffers,
+            swapchain_extent,
+            graphics_pipeline);
+
+    submit_info.sType = VK_STRUCTURE_TYPE_SUBMIT_INFO;
+    submit_info.waitSemaphoreCount = 1;
+    submit_info.pWaitSemaphores = wait_semaphores;
+    submit_info.pWaitDstStageMask = wait_stages;
+    submit_info.commandBufferCount = 1;
+    submit_info.pCommandBuffers = &command_buffer;
+    submit_info.signalSemaphoreCount = 1;
+    submit_info.pSignalSemaphores = signal_semaphores;
+
+    present_info.sType = VK_STRUCTURE_TYPE_PRESENT_INFO_KHR;
+    present_info.waitSemaphoreCount = 1;
+    present_info.pWaitSemaphores = signal_semaphores;
+    present_info.swapchainCount = 1;
+    present_info.pSwapchains = swapchains;
+    present_info.pImageIndices = &image_index;
+
+    result = vkQueueSubmit(drawing_queue, 1, &submit_info, fence_in_flight);
+    if (VK_SUCCESS != result) {
+        throw std::runtime_error("Failed to submit draw command buffer");
+    }
+
+    result = vkQueuePresentKHR(presentation_queue, &present_info);
+    if (VK_SUCCESS != result) {
+        throw std::runtime_error("Failed to queue image for presentation");
+    }
+}
+
+static VkSemaphore
+create_vulkan_semaphore(const VkDevice& logical_device)
+{
+    VkResult result;
+    VkSemaphore semaphore;
+    VkSemaphoreCreateInfo semaphore_info;
+
+    result = VK_ERROR_UNKNOWN;
+    semaphore = {};
+    semaphore_info = {};
+
+    semaphore_info.sType = VK_STRUCTURE_TYPE_SEMAPHORE_CREATE_INFO;
+
+    result = vkCreateSemaphore(logical_device, &semaphore_info, nullptr, &semaphore);
+    if (VK_SUCCESS != result) {
+        throw std::runtime_error("Failed to create vulkan semaphore");
+    }
+
+    return semaphore;
+}
+
+static VkFence
+create_vulkan_fence(const VkDevice& logical_device)
+{
+    VkResult result;
+    VkFence fence;
+    VkFenceCreateInfo fence_info;
+
+    result = VK_ERROR_UNKNOWN;
+    fence = {};
+    fence_info = {};
+
+    fence_info.sType = VK_STRUCTURE_TYPE_FENCE_CREATE_INFO;
+    fence_info.flags = VK_FENCE_CREATE_SIGNALED_BIT;
+
+    result = vkCreateFence(logical_device, &fence_info, nullptr, &fence);
+    if (VK_SUCCESS != result) {
+        throw std::runtime_error("Failed to create vulkan fence");
+    }
+
+    return fence;
+}
+
+static void
+game(void)
+{
+    std::int32_t ret;
+    std::uint32_t flags;
+
+    std::uint32_t extension_count;
+    std::vector<const char*> extension_names;
+    std::vector<VkFramebuffer> swapchain_framebuffers;
+
+    std::string msg_temp; // TODO start using g_msg_temp instead
```

Comment:
Please address TODO.

Reply:
Addressed in: https://github.com/pottumuusi/neotetris/pull/1/commits/0596f64d4ae5c59ab4c7ea7dc1d48778759d376c